### ISSUE TITLE
Text input nodes register Input Actions as text

### ISSUE THREAD

**Issue by face-hh** [godotengine/godot#88768]
### Tested versions

- Reproducible in v4.3.dev.custom_build [`0246230e2`], as well as the stable v4.2

### System information

Godot v4.3.dev (`0246230e2`) - EndeavourOS #1 SMP PREEMPT_DYNAMIC Sun, 21 Jan 2024 22:14:10 +0000 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 3GB (nvidia; 545.29.06) - Intel(R) Core(TM) i5-4670K CPU @ 3.40GHz (4 Threads)

### Issue description

Nodes used for Text Input (i.e. `CodeEdit`, `TextEdit`, `LineEdit`) should **not** register shortcuts (known as **actions**) as actual input, but rather ignore it.

Example:
Pressing a custom action named `ui_test` mapped to `CTRL` + `P` shouldn't make `CodeEdit` append *"p"* to its contents.

NOTE: this can be fixed by adding this:
```gdscript
if (k->is_action("ui_open", true)) {
    accept_event();
    return;
}
```

at `scene/gui/text_edit.cpp` > `gui_input(const Ref<InputEvent>&)` & recompiling the engine.

### Steps to reproduce

1. Create a 2D scene
2. Create a text input node, `LineEdit` for example.
3. Add an action via Project > Project Settings > Input Map (action must include an a-Z key)
4. Optionally add a script to the 2D scene, checking for the action being pressed.
5. Run the project, focus the `LineEdit`, press the custom shortcut.
6. Observe the a-Z part of the shortcut being registered

### Minimal reproduction project (MRP)
Note: The action in this MRP is `ui_test` mapped to `CTRL` + `P`

[Godot issue report.zip](https://github.com/godotengine/godot/files/14393691/Godot.issue.report.zip)

**Comment by face-hh** [comment]
Corrections:

> this can be fixed by adding this:

Modifying `core/input/input_map.cpp` > `get_builtins()` is also required. Adding the action through Godot editor won't work with the fix I previously mentioned.
```cpp
// At line ~390, in case it helps.
inputs = List<Ref<InputEvent>>();
inputs.push_back(InputEventKey::create_reference(Key::P | KeyModifierMask::CMD_OR_CTRL));
default_builtin_cache.insert("ui_test", inputs);
```
Then adding it to the `_builtin_action_display_names`:
```cpp
// At line ~280
// ...
{ "ui_test",                                       TTRC("Testing") },
// ...
```

TL;DR: action must be builtin for the solution I've mentioned to work.

**Comment by AThousandShips** [comment]
This is likely not supported, and should be documented instead id say, see https://github.com/godotengine/godot/issues/76566

**Comment by face-hh** [comment]
> This is likely not supported, and should be documented instead id say, see #76566
> 
> Though in this case I'm not even sure that matters, I don't think it's expected that the UI elements should ignore things just because it's an action, that would make you unable to enter `W` or `A` or `S` or `D` for example

I believe UI elements that receive text should at least have a method/property to make them ignore the input if it's an action.

According to my attempt to fix this via GDScript, I came to the conclusion it's not possible to prevent the `CodeEdit` from behaving this way.

It is somewhat possible by detecting when `CTRL` is hold & locking the `CodeEdit`, but that's not optimal since it also disables the usage of `CTRL` + `C`, `CTRL` + `→`, etc.

Having to recompile the engine for this isn't necessarily time consuming, but it would definitely be better if there was an option to handle this without doing so.

I agree that it should be at least documented as I couldn't find any information about this when facing the issue.

Regardless, I suppose I'll continue to use the workaround I found.

**Comment by AThousandShips** [comment]
I'd say it's expected that it doesn't process any actions it doesn't use, I'd say what you expect is not what is the default assumption, the input *isn't* an action, it's just a key, i.e. it's `InpitEventKey` not `InputEventAction`

You can probably handle this with scripts by overriding `_gui_input` and calling `accept_event`, which is the proper way of handling input, and is [documented](https://docs.godotengine.org/en/stable/tutorials/inputs/inputevent.html#how-does-it-work)

**Comment by face-hh** [comment]
> You can probably handle this with scripts by overriding `_gui_input` and calling `accept_event`, which is the proper way of handling input, and is [documented](https://docs.godotengine.org/en/stable/tutorials/inputs/inputevent.html#how-does-it-work)

Dang you're right, that worked!

Here's the working script for future reference:
```gdscript
extends Node2D
@onready var code_edit = $CodeEdit

func _process(delta):
	if Input.is_action_just_pressed("ui_test"):
		print("Hey!")


func _on_code_edit_gui_input(event):
	if !Input.is_action_just_pressed("ui_test"): return;

	code_edit.accept_event()
```

**Comment by AThousandShips** [comment]
Great it worked out!

As this is in my opinion expected and desired, and relatively niche, and the way to solve it is documented (in a few places, including `_gui_input`) I'll go ahead and close it as resolved

Thank you for your report! Good luck in your projects!
